libbeat/processors/annotate/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,163 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+
+	"github.com/ericchiang/k8s"
+	"github.com/ghodss/yaml"
+)
+
+const (
+	timeout = time.Second * 5
+)
+
+var (
+	fatalError = errors.New("Unable to create kubernetes processor")
+)
+
+type kubernetesAnnotator struct {
+	podWatcher *PodWatcher
+}
+
+type kubeAnnotatorConfig struct {
+	InCluster  bool          `config:"in_cluster"`
+	KubeConfig string        `config:"kube_config"`
+	Host       string        `config:"host"`
+	Namespace  string        `config:"namespace"`
+	SyncPeriod time.Duration `config:"sync_period"`
+}
+
+func init() {
+	processors.RegisterPlugin("kubernetes", newKubernetesAnnotator)
+
+	// Register default indexers
+	Indexing.AddIndexer(&PodNameIndexer{})
+	Indexing.AddIndexer(&ContainerIndexer{})
+}
+
+func newKubernetesAnnotator(cfg common.Config) (processors.Processor, error) {
+	config := kubeAnnotatorConfig{
```

Comment:
we normally have the defaults as a var external to the method. Something called `defaultKubeAnnotatorConfig` in a `config.go` file. Check some other `config.go` files for an example. This makes it easy to find the config options.